app/models/illumina_htp/initial_stock_tube_purpose.rb

```diff
@@ -1,36 +1,12 @@
+# frozen_string_literal: true
+
 # Class initially used to represent the first tube in a pipeline following pooling
 # from a plate. However appears to be a little more widespread in Limber pipelines.
 # eg. LB Custom Pool Norm
+# In fact, in the limber pipeline it appears to be responsible for any intermediate
+# tubes.
 class IlluminaHtp::InitialStockTubePurpose < IlluminaHtp::StockTubePurpose
-  def valid_transition?(outer_request, target_state)
```

Comment:
I've extracted state change logic into the state changers. I should probably do the same for the state calculators.